server/routes/CheckRoleMiddleware.js

```diff
@@ -0,0 +1,10 @@
+function checkRole(roleArray) {
+    return function(req, res, next) {
+      if (req.user && roleArray.includes(req.user.role.name)) {
+        next();
+      } else {
+        res.status(403).send('Not authorized');
```

Comment:
This is a great example for HandledError. You can replace this line with `throw new HandledError(403, "Not authorized");` if you want.

Reply:
Thanks, noted